app/src/main/java/com/example/musictoned/editRoutine/EditRoutineScreen.kt

```diff
@@ -90,27 +88,26 @@ import java.util.Locale
 @Composable
 fun EditRoutineScreen(
     onNavigateToRoutine: (routineID: Int?) -> Unit,
-    onNavigateToAddExercise: () -> Unit,
     routineID: Int?
 ) {
-    if(routineID != null){
-        println(AllWorkouts.getWorkout(routineID))
+    val workout: MutableState<Workout> = if (LocalInspectionMode.current) {
+        // Preview mode
+        remember { mutableStateOf(Workout("New Workout")) }
+    } else if (routineID != null) {
+        // Existing workout
+        remember { mutableStateOf(AllWorkouts.getWorkout(routineID)) }
+    } else {
+        // New workout
+        remember { mutableStateOf(AllWorkouts.getWorkoutInProgress()) }
     }
-    //var workout by remember { mutableStateOf(AllWorkouts.getWorkout(routineID ?: 0)) }
-    var workout by remember { mutableStateOf(AllWorkouts.getWorkoutInProgress()) }
-
-    if (routineID != null){
-        workout = AllWorkouts.getWorkout(routineID)
-    }
-
 
     var popupControl by remember { mutableStateOf(false) }
 
-    var exercises = remember { mutableStateListOf<WorkoutExercise>() }
+    val exercises = remember { mutableStateListOf<WorkoutExercise>() }
 
-    var swapped = remember { mutableStateOf(false) }
+    val swapped = remember { mutableStateOf(false) }
 
-    exercises.addAll(workout.exercises.toList())
+    exercises.addAll(workout.component1().exercises.toList())
```

Comment:
what is the purpose of component1()?

Reply:
The `workout` object changed to a `MutableState<Exercise>` object, so in order to touch the contained object you need to call `.componentX()`

Reply:
See line 93